status-im/status-react
# Fix #1886

There were multiple problems with previous approach of always
expecting explicit "sent" signal from async handler, otherwise
displaying the `/send` message as pending, the most serious
problem was that the command handler is only called once during
send time, and it's problematic to depend on the result of that
on the recepient side, which could be for example offline.

The safer approach is to get rid of status code dependent display
at all and just not send the message for further processing if async
handler returns result code other then `success`.
The code is cleaner as well.
Handler-data path is actually not used anymore, but I decide not to 
delete it for now, as I think it could be beneficial in future for different
kind of handlers then simple `/send` command.

fixes #1886

### Steps to test:
- Open Status
- Open 1-1 chat and send transaction
- Check the messages on the sender/receiver side

[comment]: # (PRs will only be accepted if squashed into single commit.)
status: ready
